harpy/scripts/demultiplex_gen1.py

```
@@ -0,0 +1,168 @@
+import pysam
+from Levenshtein import distance
+
+def read_barcodes(file_path, segment):
+    """Read and parse input barcode (segment) file of segment<tab>sequence"""
+    data_dict = {}
+    with open(file_path, 'r') as file:
+        for line in file:
+            try:
+                code, seq = line.rstrip().split()
+                if code[0].upper() != segment:
+                    parser.error(f"Segments in {file_path} are expected to begin with {segment}, but begin with {code[0].upper()}")
```

Ruff (0.8.2)

Undefined name `parser`

(F821)